src/componets/CalculatorUI.js

```diff
@@ -2,37 +2,53 @@
 /* eslint-disable no-useless-constructor */
 import React from 'react';
 import './calculatorUI.css';
+import calculate from '../logic/calculate';
 
 class CalculatorUI extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((prevState) => calculate(prevState, e.target.innerText));
   }
 
```

Comment:
[Optional]
- [ ] For this milestone, we are expecting you to practice what you have learned until now; like passing props to child components, using lifecycle methods, lifting state up. You should use these concepts in your app.

Tip:
_You should divide your `Calculator` component into child components in order to practice these concepts. (For example, you can create a display component for displaying calculations and a keypad component for the keys.)_



From the project requirements ⬇️ 
>Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift the state up.